### PR TITLE
Convert Group to suspend functions

### PR DESCRIPTION
**Pull request by Dominaezzz**
Signed-off-by: Dominic Fischer <dominicfischer7@gmail.com>

### Pull Request Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [x] Pull request is based on the develop branch
- [ ] Pull request updates [CHANGES.md](https://github.com/vector-im/element-android/blob/develop/CHANGES.md)
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)

### CODE REVIEW

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/group/DefaultGroup.kt [hunk 1]
```diff
@@ -16,20 +16,13 @@
...
-                            private val taskExecutor: TaskExecutor,
                             private val getGroupDataTask: GetGroupDataTask) : Group {
 
-    override fun fetchGroupData(callback: MatrixCallback<Unit>): Cancelable {
+    override suspend fun fetchGroupData() {
         val params = GetGroupDataTask.Params.FetchWithIds(listOf(groupId))
-        return getGroupDataTask.configureWith(params) {
-            this.callback = callback
-        }.executeBy(taskExecutor)
+        return getGroupDataTask.execute(params)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/group/DefaultGroup.kt:26]: `return` is needless :)

**Review by bmarty — COMMENTED** [review]
one remark

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/group/DefaultGroup.kt (hunk 1 above)
Comment (Dominaezzz) [reply matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/group/DefaultGroup.kt:26]: Oops :sweat_smile:

**Review by bmarty — APPROVED** [review]
Thanks for the update. LGTM.

**Pull request merged**